### DISCUSSION TITLE
Friction coeff

### DISCUSSION THREAD

**Discussion by limbryan** [dartsim/dart#2226] [Q&A] [answered]
Hi, 

I am trying to change the friction between a walking quadruped and and the floor created. I have searched quite extensively for how to do this but have not found the exact way of doing it. I am currently getting the skeleton and Bdoy Node of the ground and setting the friction using setFrictionCoeff and likewise getting the skeleton and Body node of my robot and setting that friction as well. Is this the only way of doing it? Also, i notice the values here refer to the maximum tangential force and not really the friction coefficient that is commonly used that lies between 0 and 1. Is there a way to do this? Thanks

**Comment by jslee02** [comment, marked as answer]
Sorry this sat unanswered for so long. This may no longer be needed by the original poster, but for anyone else with a similar question:

For DART 6.x, including the 6.8-era API, setting the friction on the contacting bodies was the intended way to affect contact friction. In current DART 7/main, the same idea lives on each colliding `ShapeNode`'s `DynamicsAspect` via `setFrictionCoeff()`, so setting it on the relevant ground and robot collision shape nodes is still the right approach.

The value is a Coulomb friction coefficient, not an absolute maximum tangential force by itself. The contact solver uses it as the tangential friction bound coupled to the normal contact row, so the effective tangential limit depends on the normal contact impulse/force.

DART accepts non-negative friction coefficients; they are not limited to `[0, 1]`, so values above 1 can be valid for high-friction contacts.

One detail to keep in mind: when two surfaces contact, DART combines the two surface coefficients conservatively using the smaller value. If one side still has a lower coefficient, increasing only the other side may not change the resulting contact friction.
